### ISSUE TITLE
Incorrect linter reports all final vars need init when using static

### ISSUE THREAD

**Issue by moneer-muntazah** [dart-lang/linter#3168]
If you try the dartplayground with 

```dart 
class Foo {
  Foo();
  static final String? _somevar;
}
```

You will net a lint on the constructor that says that all final variables need to be initialized but _somevar isn't. This should be on the class rather than the constructor. In Android Studio the fix even adds `this._somevar` in the constructor.
